python/cudf/cudf/core/dtypes.py

```diff
@@ -64,3 +73,19 @@ def __eq__(self, other):
 
     def construct_from_string(self):
         raise NotImplementedError()
+
+    def serialize(self):
+        header = {}
+        frames = []
+        header["ordered"] = self.ordered
+        categories_buffer = []
+        if self.categories is not None:
+            categories_buffer = [self.categories.as_column()._data_view()]
```

Comment:
Should defer to `self.categories.serialize` instead of reaching directly into memory here. What if categories is a string column?

Reply:
Fixed